lib/hermann/provider/java_producer.rb

```diff
@@ -43,7 +43,7 @@ def initialize(brokers, opts={})
       # @return +Concurrent::Promise+ Representa a promise to send the
       #   data to the kafka broker.  Upon execution the Promise's status
       #   will be set
-      def push_single(msg, topic, key)
+      def push_single(msg, topic, key, _)
```

Comment:
Is there a benefit to modifying the Java producer's signature? They were already different, so I'm not sure why bother changing this